x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/azureopenai/embeddings/AzureOpenAiEmbeddingsServiceSettings.java

```diff
@@ -151,6 +175,7 @@ public AzureOpenAiEmbeddingsServiceSettings(StreamInput in) throws IOException {
         dimensionsSetByUser = in.readBoolean();
         maxInputTokens = in.readOptionalVInt();
         similarity = in.readOptionalEnum(SimilarityMeasure.class);
+        rateLimitSettings = RateLimitSettings.of(in, DEFAULT_RATE_LIMIT_SETTINGS);
```

Comment:
I use the same pattern for all the external services. Let me know what you think, the `if-block` check for the transport version is within the `of()` call. Maybe that's not a good idea because it hides it from the constructors here? 🤷‍♂️

Reply:
I think this makes sense the way it is. Since `of` takes a StreamInput parameter, it makes sense that it would be doing the versioning.

Reply:
> This is pattern is more readable but harder to understand.

Yeah good point. I'll switch it back to the `if-blocks` in the constructors.